crates/ruff/src/rules/flynt/helpers.rs

```diff
@@ -0,0 +1,64 @@
+use ruff_python_ast::helpers::create_expr;
+use rustpython_parser::ast::{Constant, Expr, ExprKind};
+
+fn to_formatted_value_expr(inner: Expr) -> Expr {
+    create_expr(ExprKind::FormattedValue {
+        value: Box::new(inner),
+        conversion: 0,
+        format_spec: None,
+    })
+}
+
+/// Figure out if `expr` represents a "simple" call
+/// (i.e. one that can be safely converted to a formatted value).
+fn is_simple_call(expr: &Expr) -> bool {
+    match &expr.node {
+        ExprKind::Call {
+            func,
+            args,
+            keywords,
+        } => args.is_empty() && keywords.is_empty() && is_simple_callee(func),
+        _ => false,
+    }
+}
+
+/// Figure out if `func` represents a "simple" callee (a bare name, or a chain of simple
+/// attribute accesses).
+fn is_simple_callee(func: &Expr) -> bool {
+    match &func.node {
+        ExprKind::Name { .. } => true,
+        ExprKind::Attribute { value, .. } => is_simple_callee(value),
+        _ => false,
+    }
+}
+
+/// Convert an expression to a f-string element (if it looks like a good idea).
+pub fn to_fstring_elem(expr: Expr) -> Option<Expr> {
```

Comment:
I recommend taking the `Expr` as a reference and cloning it on demand, or (a little more complicated) to return an `Option<Cow<Expr>>` to avoid cloning the `Expr` altogether.